### PR TITLE
Flexible inspector properties

### LINKED ISSUE #3990: [Inspector] Details section should be flexible in properties to display
## Overview
The details section of the inspector only allows displaying properties of a known installed type, when it should be more flexible. A use case would be displaying different properties for different selections in views.

## Expected Behavior
Details section should display custom properties if desired, or the standard properties otherwise

## Steps to Reproduce (before)
1. select an object and see that it displays properties based on the object's type
2. select a layout object within a display layout and see there are no properties

## Steps to Reproduce (after)
1. select an object with passed in custom details and see that the custom details are displayed, regardless of installed type

### PR DESCRIPTION
**Pull request by davetsay**
Closes #3990 

### All Submissions:

* [x] Have you followed the guidelines in our [Contributing document](https://github.com/nasa/openmct/blob/master/CONTRIBUTING.md)?
* [x] Have you checked to ensure there aren't other open [Pull Requests](https://github.com/nasa/openmct/pulls) for the same update/change?

### Author Checklist

* [x] Changes address original issue?
* [ ] Unit tests included and/or updated with changes?
* [ ] Command line build passes? pending non related lint fixes in #4082
* [x] Has this been smoke tested?
* [x] Testing instructions included in associated issue?

### CODE REVIEW

**Comment by davetsay** [comment]
@akhenry this is the change which allows us to customize the details block in inspector, and specifically custom details for MMGIS features.

Writing tests now but wanted you to verify this strategy first if possible

**Review by akhenry — CHANGES_REQUESTED** [review]
Looks good to me! Just two questions.

File: src/ui/inspector/Inspector.vue [hunk 1]
```diff
@@ -19,7 +41,9 @@
                    type="vertical"
         >
             <pane class="c-inspector__properties">
-                <properties />
+                <Properties
+                    v-if="!activity"
```
[the remark this answers is not in the document]
Comment (davetsay) [reply src/ui/inspector/Inspector.vue:45]: I don't know to be honest. I just bubbled this logic up from the properties component. I'll take another look.

File: src/ui/inspector/details/Properties.vue [hunk 2]
```diff
@@ -0,0 +1,199 @@
...
+<template>
+<div class="c-inspector__properties c-inspect-properties">
+    <div class="c-inspect-properties__header">
+        Details
+    </div>
+    <ul
+        v-if="hasDetails"
+        class="c-inspect-properties__section"
+    >
+        <Component
```
[the remark this answers is not in the document]
Comment (davetsay) [reply src/ui/inspector/details/Properties.vue:32]: removed `detail-time`

File: src/ui/inspector/Inspector.vue (hunk 1 above)
[the remark this answers is not in the document]
Comment (davetsay) [reply src/ui/inspector/Inspector.vue:45]: Created #4180 to refactor plan activities to leverage this code so the hard coding can be removed. It looks like we can do so but need to check with someone familiar with plan design to make sure (@charlesh88 , @tiffanyatruong , @shefalijoshi).

[the remark this answers is not in the document]
Comment (akhenry) [reply src/ui/inspector/Inspector.vue:45]: Sounds good, I'm happy for this to be addressed in a followup if this is pre-existing logic.

**Review by akhenry — CHANGES_REQUESTED** [review]
Just needs some unit tests, otherwise it's good to go. You can mount and test the component in isolation, no need to test it in the context of the full app.

**Comment by davetsay** [comment]
> Just needs some unit tests, otherwise it's good to go. You can mount and test the component in isolation, no need to test it in the context of the full app.

Added unit tests

**Review by akhenry — APPROVED** [review]
Looks great. Thanks!

### Reviewer Checklist

1. Changes appear to address issue? Y
2. Appropriate unit tests included? Y
3. Code style and in-line documentation are appropriate? Y
4. Commit messages meet standards? Y
5. Has associated issue been labelled `unverified`? (only applicable if this PR closes the issue) Y

**Pull request merged**